src/kOS/Sound/VoiceValue.cs

```diff
@@ -0,0 +1,149 @@
+            this.maker = maker;
+            this.updateHandler = updateHandler;
+            voice = maker.GetVoice(voiceNum);
+            maker.SetWave(voiceNum, "square");
+            this.updateHandler.AddObserver(this);
+
+            InitalizeSuffixes();
+        }
+        
+        public void InitalizeSuffixes()
+        {
+            AddSuffix("ATTACK", new SetSuffix<ScalarValue>(() => voice.Attack, value => voice.Attack = value));
+            AddSuffix("DECAY", new SetSuffix<ScalarValue>(() => voice.Decay, value => voice.Decay = value));
+            AddSuffix("SUSTAIN", new SetSuffix<ScalarValue>(() => voice.Sustain, value => voice.Sustain = value));
+            AddSuffix("RELEASE", new SetSuffix<ScalarValue>(() => voice.Release, value => voice.Release = value));
+            AddSuffix("VOLUME", new SetSuffix<ScalarValue>(() => voice.Volume, value => voice.Volume = value));
+            AddSuffix("WAVE", new SetSuffix<StringValue>(() => maker.GetWaveName(voiceNum), value => maker.SetWave(voiceNum, value.ToString())));
+            AddSuffix("PLAY", new OneArgsSuffix<Structure>(Play));
+            AddSuffix("LOOP", new SetSuffix<BooleanValue>(() => loop, value => loop = value));
+            AddSuffix("ISPLAYING", new SetSuffix<BooleanValue>(() => IsPlaying, value => IsPlaying = value));
+            AddSuffix("TEMPO", new SetSuffix<ScalarValue>(() => tempo, value => tempo = (float)value.GetDoubleValue()));
+        }
+
+        public void Play(Structure notes)
+        {
+            if (notes is NoteValue)
+            {
+                song = new ListValue();
+                song.Add(notes);
+            }
+            else if (notes is ListValue)
+            {
+                song = notes as ListValue;
+            }
+            else
+                throw new KOSInvalidArgumentException("Play", "note", "Requires either a NOTE() or a LIST() of NOTE()'s as its parameter.");
+
+            noteNum = -1;
+            noteEndTimeStamp = -1f;
+            IsPlaying = true;
+
+        }
+        
+        public void KOSUpdate(double deltaTime)
+        {
+            if (! IsPlaying)
+                return;
+
+            float now = Time.fixedTime;
+            
+            // If still playing prev note, do nothing except maybe change
+            // its frequency if it's a slidenote:
+            if (now < noteEndTimeStamp)
+            {
+                NoteValue note = song[noteNum] as NoteValue;
+                if (noteFreqTotalChange != 0.0)
+                {
+                    float durationPortion = (now - noteStartTimeStamp) / (noteEndTimeStamp - noteStartTimeStamp);
+                    float newFreq = note.Frequency + durationPortion*noteFreqTotalChange;
+                    voice.ChangeFrequency(newFreq);
+                }
+                return;
+            }
+            
+            // Increment to next note and start playing it:
+            ++noteNum;
+            if (noteNum > song.Count())
```

Comment:
It looks like you're right, but what's weird is that I've used this code to make looping music and it worked just fine.  Let me look into why.